src/geometry/geom2/transform.js

```diff
@@ -0,0 +1,32 @@
+const mat4 = require('../../math/mat4')
+
+const create = require('./create')
+
+/**
+ * Transform the given geometry using the given matrix.
+ * This is a lazy transform of the sides, as this function only adjusts the transforms.
+ * See canonicalize() for the actual application of the transfroms to the sides.
```

Comment:
wow ok, so with regard to my points above about baseSides: so canonicalize is needed to apply transforms ? I get the intention but that is very confusing to be honest.

Reply:
canonicalize was merging 'close enough' points, I do not see how applying transforms is related ?